snippets/legv8.json

```diff
@@ -0,0 +1,208 @@
+      "LDUR $reg3, [SP, #16] // restore $reg3",
+      "LDUR $reg2, [SP, #24] // restore $reg2",
+      "LDUR $reg1, [SP, #32] // restore $reg1",
+      "LDUR LR, [SP, #40] // restore LR",
+      "ADDI SP, SP, #48  // restore previous space on stack"
+    ],
+    "description": "use this to save 6 registers in the Stack, including the return address of the procedure (where the procedure was called)"
+  },
+  "stack manipulation with 7 registers": {
+    "prefix": "st7",
+    "body": [
+      "// storing registers in stack before changing them",
+      "SUBI SP, SP, #56 // make room on stack (SP) for 2 registers",
+      "STUR LR, [SP, #48] // store return adress (LR) of procedure",
+      "STUR $reg1, [SP, #40] // store $reg1",
+      "STUR $reg2, [SP, #32] // store $reg2",
+      "STUR $reg3, [SP, #24] // store $reg3",
+      "STUR $reg4, [SP, #16] // store $reg4",
+      "STUR $reg5, [SP, #8] // store $reg5",
+      "STUR $reg6, [SP, #0] // store $reg6",
+      "// body of the procedure",
+      "$body",
+      "// restoring saved registers from stack",
+      "LDUR $reg6, [SP, #0] // restore $reg6",
+      "LDUR $reg5, [SP, #8] // restore $reg5",
+      "LDUR $reg4, [SP, #16] // restore $reg4",
+      "LDUR $reg3, [SP, #24] // restore $reg3",
+      "LDUR $reg2, [SP, #32] // restore $reg2",
+      "LDUR $reg1, [SP, #40] // restore $reg1",
+      "LDUR LR, [SP, #48] // restore LR",
+      "ADDI SP, SP, #56  // restore previous space on stack"
+    ],
+    "description": "use this to save 7 registers in the Stack, including the return address of the procedure (where the procedure was called)"
+  },
+  "stack manipulation with 8 registers": {
+    "prefix": "st8",
+    "body": [
+      "// storing registers in stack before changing them",
+      "SUBI SP, SP, #64 // make room on stack (SP) for 2 registers",
+      "STUR LR, [SP, #56] // store return adress (LR) of procedure",
+      "STUR $reg1, [SP, #48] // store $reg1",
+      "STUR $reg2, [SP, #40] // store $reg2",
+      "STUR $reg3, [SP, #32] // store $reg3",
+      "STUR $reg4, [SP, #24] // store $reg4",
+      "STUR $reg5, [SP, #16] // store $reg5",
+      "STUR $reg6, [SP, #8] // store $reg6",
+      "STUR $reg7, [SP, #0] // store $reg7",
+      "// body of the procedure",
+      "$body",
+      "// restoring saved registers from stack",
+      "LDUR $reg7, [SP, #0] // restore $reg7",
+      "LDUR $reg6, [SP, #8] // restore $reg6",
+      "LDUR $reg5, [SP, #16] // restore $reg5",
+      "LDUR $reg4, [SP, #24] // restore $reg4",
+      "LDUR $reg3, [SP, #32] // restore $reg3",
+      "LDUR $reg2, [SP, #40] // restore $reg2",
+      "LDUR $reg1, [SP, #48] // restore $reg1",
+      "LDUR LR, [SP, #56] // restore LR",
+      "ADDI SP, SP, #64  // restore previous space on stack"
+    ],
+    "description": "use this to save 8 registers in the Stack, including the return address of the procedure (where the procedure was called)"
+  },
+  "incremental for loop": {
```

Comment:
A few suggestions for the loop. 
- `ifl` isn't immediately understandable so being a bit more explicit makes it easier to grasp and remember.
- The use of a temporary register is annoying when writing legv8 code as it requires use of another register. Using flags, this can be simplified.
- Also not using the term `arr_len` and using a generic upper limit makes it clearer that this can be used for any need, not just arrays.
- I've also modified the description to include the equivalent C code.

```json
  "increasing for loop": {
    "prefix": "for_increasing",
    "body": [
      "${1:loop name}_loop:",
      "// if counter ($2) greater than or equal to limit ($3), then exit",
      "SUBS XZR, ${2:counter}, ${3:upper_limit}",
      "B.GE ${1}_loop_done",
      "// loop body",
      "",
      "// increment counter ($2) by $4",
      "ADDI $2, $2, #${4:increment_amount}",
      "B $1_loop",
      "$1_loop_done:"
    ],
    "description": "Creates a for loop from counter to upper_limit. for (; counter < upper_limit; counter += increment_amount); Note: the counter and array size must be initialized before the loop."
  },
 ```

Reply:
I agree, It`s Implemented